magpie/setup/magpie-setup-project-hive

```diff
@@ -0,0 +1,106 @@
+#  traditional HPC systems.  For details, see https://github.com/llnl/magpie.
+#  
+#  Magpie is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+#  
+#  Magpie is distributed in the hope that it will be useful, but
+#  WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  General Public License for more details.
+#  
+#  You should have received a copy of the GNU General Public License
+#  along with Magpie.  If not, see <http://www.gnu.org/licenses/>.
+#############################################################################
+
+# This script sets up configuration files for jobs.  For the most
+# part, it shouldn't be editted.  See job submission files for
+# configuration details.
+
+if [ "${HIVE_SETUP}" != "yes" ]
+then
+    exit 0
+fi
+
+source ${MAGPIE_SCRIPTS_HOME}/magpie/exports/magpie-exports-dirs
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-node-identification
+source ${MAGPIE_SCRIPTS_HOME}/magpie/lib/magpie-lib-setup
+
+hiveconfdirsubst=`echo "${HIVE_CONF_DIR}" | sed "s/\\//\\\\\\\\\//g"`
+hivemetastoredirsubst=`echo "${HIVE_METASTORE_DIR}" | sed "s/\\//\\\\\\\\\//g"`
+hivedefaultmetastoredirsubst=`echo "${HIVE_DEFAULT_METASTORE_DIR}" | sed "s/\\//\\\\\\\\\//g"`
+
+# zeppelinnoderank set if succeed
+if ! Magpie_am_I_a_hive_node
+then
+    exit 0
+fi
+ 
+# For rest of setup, we will use cluster specific paths
+Magpie_make_all_local_dirs_node_specific
+
+#
+# Get config files for setup
+#
+
+# Magpie_find_conffile will set the 'pre' filenames
+
+Magpie_find_conffile "Hive" ${HIVE_CONF_FILES:-""} "hive-site.xml" "pre_hiveproperties"
+Magpie_find_conffile "Hive" ${HIVE_CONF_FILES:-""} "hive-default.xml.template" "pre_hivedefaultproperties"
+#Magpie_find_conffile "Hive" ${HIVE_CONF_FILES:-""} "hivemetastore-site.xml" "pre_hivemetastoreproperties"
+Magpie_find_conffile "Hive" ${HIVE_CONF_FILES:-""} "hive.log4j.properties" "pre_log4jproperties"
+Magpie_find_conffile "Hive" ${HIVE_CONF_FILES:-""} "hive-env.sh" "pre_hiveenvsh"
+
+post_hiveproperties=${HIVE_CONF_DIR}/hive-site.xml
+post_hivedefaultproperties=${HIVE_CONF_DIR}/hive-default.xml
+#post_hivemetastoreproperties=${HIVE_CONF_DIR}/hivemetastore-site.xml
+post_log4jproperties=${HIVE_CONF_DIR}/log4j.properties
+post_hiveenvsh=${HIVE_CONF_DIR}/hive-env.sh
+
+post_hivesparkproperties=${SPARK_CONF_DIR}/hive-site.xml
+#post_hivesparkmetastoreproperties=${SPARK_CONF_DIR}/hivemetastore-site.xml
+
+# Setup Hive configuration files and environment files
+cp ${pre_hiveproperties} ${post_hiveproperties}
+cp ${pre_hiveproperties} ${post_hivesparkproperties}
+
+# Add the zookeeper quorum to hive-site.xml
+quorumlist=`awk -vORS=, '{ print $1,":2181" }' ${ZOOKEEPER_CONF_DIR}/slaves | sed 's/ //g' | sed 's/,$/\n/' `
+
```

Comment:
does hive need zookeeper?  Should it be added as a dependency in other code?  (magpie-check-inputs, hive-run code above, etc.)

Reply:
Agreed/changed.

Reply:
Ahh ok.   I suppose we can either make it a requirement (within the magpie world) or an option going forward.  The README.hive seems to list it as a requirement.  What would you think is better?